src/ContentfulImage.js

```diff
@@ -0,0 +1,126 @@
+import React, { forwardRef, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import buildUrlParameters from './utils';
+import { FORMAT, RESIZE, CROP, QUALITY, BACKGROUND_COLOR } from './constants';
+
+const ContentfulImage = forwardRef(({ format, optimize, resize, cropRadius, quality, backgroundColor, ...imageProps }, ref) => {
+    const { convertedUrl, mimeType, originalUrl } = useMemo(() => {
+        const originalUrl = imageProps.src;
+        const urlParameters = [];
+        let returnedMimeType;
+
+        if (optimize) {
+            const { url, mimeType } = buildUrlParameters({
+                key: FORMAT,
+                value: format,
+                includeMimeType: true,
+            });
+
+            urlParameters.push(...url);
+            returnedMimeType = mimeType;
+        }
+
+        if (resize) {
+            const url = buildUrlParameters({
+                key: RESIZE,
+                value: resize,
+            });
+
+            urlParameters.push(...url);
+        }
+
+        if (cropRadius) {
+            const url = buildUrlParameters({
+                key: CROP,
+                value: cropRadius,
+            });
+
+            urlParameters.push(...url);
+        }
+
+        if (quality) {
+            const url = buildUrlParameters({
+                key: QUALITY,
+                value: quality,
+            });
+
+            urlParameters.push(...url);
+        }
+
+        if (backgroundColor) {
+            const url = buildUrlParameters({
+                key: BACKGROUND_COLOR,
+                value: backgroundColor,
+            });
+
+            urlParameters.push(...url);
+        }
+
+        const convertedUrl = urlParameters.length ?
+            `${originalUrl}?${urlParameters.join('&')}` :
+            originalUrl;
+
+        return {
+            convertedUrl,
```

Comment:
Nitpick: maybe we should have a better name for this `convertedUrl`. Something that would demonstrate that it is a url with augmented parameters to optimize the image.

Maybe `enhancedUrl` ? Don't know 🤔